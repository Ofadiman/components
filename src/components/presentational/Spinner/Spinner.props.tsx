import { ComponentStoriesProps } from '../../../types/ComponentStoriesProps.type'
import { SpinnerProps } from './Spinner.types'

export const spinnerStoryProps: ComponentStoriesProps<SpinnerProps> = [
  { size: 'regular', title: 'Default' },
  { size: 'small', title: 'Small' },
  { size: 'large', title: 'Large' }
]
