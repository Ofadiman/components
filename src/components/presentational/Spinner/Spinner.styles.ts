import { makeStyles } from '@material-ui/core/styles'

export const useSpinnerStyles = makeStyles(
  () => ({
    '@keyframes spinnerAnimation': {
      from: {
        transform: 'rotate(0)'
      },
      to: {
        transform: 'rotate(360deg)'
      }
    },
    arm: {
      '&:nth-child(1)': {
        animation: '$spinnerAnimation 2s ease-in-out infinite',
        borderColor: '#99ff33'
      },
      '&:nth-child(2)': {
        animation: '$spinnerAnimation 2.5s ease-in-out infinite',
        borderColor: '#9933ff'
      },
      '&:nth-child(3)': {
        animation: '$spinnerAnimation 3s ease-in-out infinite',
        borderColor: '#cc99ff'
      },
      borderRadius: '50%',
      boxSizing: 'border-box',
      height: '100%',
      margin: 0,
      position: 'absolute',
      width: '100%'
    },
    large: {
      '& > div': {
        borderTop: '25px solid'
      },
      height: '250px',
      width: '250px'
    },
    medium: {
      '& > div': {
        borderTop: '15px solid'
      },
      height: '150px',
      width: '150px'
    },
    root: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
      position: 'relative'
    },
    small: {
      '& > div': {
        borderTop: '5px solid'
      },
      height: '50px',
      width: '50px'
    }
  }),
  {
    name: 'Spinner'
  }
)
