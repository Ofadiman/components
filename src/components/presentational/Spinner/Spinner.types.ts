export interface SpinnerProps {
  className?: string
  size?: 'small' | 'regular' | 'large'
}
