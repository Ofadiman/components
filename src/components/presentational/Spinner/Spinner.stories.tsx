import { storiesOf } from '@storybook/react'
import React from 'react'

import { Spinner } from './Spinner'
import { spinnerStoryProps } from './Spinner.props'

const clientApi = storiesOf('Components/Presentational/Spinner', module)

spinnerStoryProps.forEach(({ title, ...props }) => {
  clientApi.add(title, () => <Spinner {...props} />)
})
