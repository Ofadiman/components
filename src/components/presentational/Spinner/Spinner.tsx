import clsx from 'clsx'
import React from 'react'

import { useSpinnerStyles } from './Spinner.styles'
import { SpinnerProps } from './Spinner.types'

export const Spinner: FC<SpinnerProps> = ({ className, size = 'regular' }) => {
  const classes = useSpinnerStyles()

  return (
    <div
      className={clsx(classes.root, className, {
        [classes.small]: size === 'small',
        [classes.regular]: size === 'regular',
        [classes.large]: size === 'large'
      })}
    >
      <div className={classes.arm} />
      <div className={classes.arm} />
      <div className={classes.arm} />
    </div>
  )
}
