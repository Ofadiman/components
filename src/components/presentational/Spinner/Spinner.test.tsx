import React from 'react'

import { renderComponent } from '../../../functions/renderComponent/renderComponent'
import { Spinner } from './Spinner'
import { spinnerStoryProps } from './Spinner.props'

describe('Spinner component', () => {
  spinnerStoryProps.forEach(({ title, ...props }) => {
    test(`match snapshot for ${title}`, () => {
      const { asFragment } = renderComponent(<Spinner {...props} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
