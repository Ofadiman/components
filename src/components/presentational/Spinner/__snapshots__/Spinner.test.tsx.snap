// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component match snapshot for Default 1`] = `
<DocumentFragment>
  <div
    class="Spinner-root-5 Spinner-regular-4"
  >
    <div
      class="Spinner-arm-2"
    />
    <div
      class="Spinner-arm-2"
    />
    <div
      class="Spinner-arm-2"
    />
  </div>
</DocumentFragment>
`;

exports[`Spinner component match snapshot for Large 1`] = `
<DocumentFragment>
  <div
    class="Spinner-root-17 Spinner-large-15"
  >
    <div
      class="Spinner-arm-14"
    />
    <div
      class="Spinner-arm-14"
    />
    <div
      class="Spinner-arm-14"
    />
  </div>
</DocumentFragment>
`;

exports[`Spinner component match snapshot for Small 1`] = `
<DocumentFragment>
  <div
    class="Spinner-root-11 Spinner-small-12"
  >
    <div
      class="Spinner-arm-8"
    />
    <div
      class="Spinner-arm-8"
    />
    <div
      class="Spinner-arm-8"
    />
  </div>
</DocumentFragment>
`;
